### PR TITLE
Typefy queries

### PR DESCRIPTION
**Pull request by manzoorwanijk**
Completes #2088 

## Checklist

* [ ] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [ ] User input is adequately validated and sanitized
* [ ] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [ ] My code is tested.
* [ ] My code follows the Event Espresso code style.
* [ ] My code has proper inline documentation.
* [ ] My code accounts for when the site is in Maintenance Mode (MM2 especially disallows usage of models)

### CODE REVIEW

File: assets/prototype/domain/eventEditor/data/queries/datetimes/useDatetimeIds.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import useDatetimes from './useDatetimes';
+import { Datetime, EntityId } from '../../types';
+
+const useDatetimeIds = (): EntityId[] => {
+	const datetimes: Datetime[] = useDatetimes();
+
+	return datetimes.map(({ id }: Datetime) => id);
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/data/queries/datetimes/useDatetimeIds.ts]: We don't need `: Datetime` part here, it is aware of the underlying data:

image

File: assets/prototype/domain/eventEditor/data/queries/datetimes/useDatetimeItem.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	const queryOptions: ReadQueryOptions = {
+		query: GET_DATETIME,
+		variables: {
+			id,
+		},
+	};
+
+	try {
+		data = client.readQuery(queryOptions);
+		datetime = propOr<Datetime, string, any>(null, 'datetime', data);
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/data/queries/datetimes/useDatetimeItem.ts]: That is PRO :+1:

File: assets/prototype/domain/eventEditor/data/queries/priceTypes/useDefaultPriceType.ts [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
 /**
  * A custom react hook for retrieving the default price type object.
  */
-const useDefaultPriceType = () => {
-	const allPriceTypes = usePriceTypes();
-	const percentSurchargePriceTypes = allPriceTypes.filter(
+const useDefaultPriceType = (): PriceType => {
+	const allPriceTypes: PriceType[] = usePriceTypes();
+	const percentSurchargePriceTypes: PriceType[] = allPriceTypes.filter(
 		({ baseType, isPercent }) => baseType === 'SURCHARGE' && isPercent
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/data/queries/priceTypes/useDefaultPriceType.ts]: This is dope:
image

File: assets/prototype/domain/eventEditor/data/queries/datetimes/useDatetimeIds.ts (hunk 1 above)
Comment (manzoorwanijk) [reply assets/prototype/domain/eventEditor/data/queries/datetimes/useDatetimeIds.ts]: true

File: assets/prototype/domain/eventEditor/data/queries/priceTypes/usePriceTypeForPrice.ts [hunk 4]
```diff
@@ -1,28 +1,28 @@
...
 		entity: 'prices',
 		entityId: priceId,
 		relation: 'priceTypes',
 	});
 
 	// get the default price type object
-	const defaultPriceType = useDefaultPriceType();
-	const relatedPriceTypes = usePriceTypes(relatedPriceTypeIds);
+	const defaultPriceType: PriceType = useDefaultPriceType();
+	const relatedPriceTypes: PriceType[] = usePriceTypes(relatedPriceTypeIds);
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/data/queries/priceTypes/usePriceTypeForPrice.ts]: TS is smart enough to detect some underlying logic:
image

**Pull request merged**